ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/TransactionRejector.scala

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.kvutils.committer.transaction
+
+import com.codahale.metrics.Counter
+import com.daml.ledger.participant.state.kvutils.DamlKvutils.{
+  DamlTransactionRejectionEntry,
+  Disputed,
+  Inconsistent,
+  InvalidLedgerTime,
+  PartyNotKnownOnLedger,
+  ResourcesExhausted,
+  SubmitterCannotActViaParticipant,
+}
+import com.daml.ledger.participant.state.kvutils.committer.Committer.buildLogEntryWithOptionalRecordTime
+import com.daml.ledger.participant.state.kvutils.committer.{StepResult, StepStop}
+import com.daml.ledger.participant.state.v1.RejectionReasonV0
+import com.daml.lf.data.Time.Timestamp
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
+import com.daml.metrics.Metrics
+
+private[transaction] class TransactionRejector(metrics: Metrics) {
```

Comment:
I find the naming a bit odd, i.e., this class seems to be rather a factory of `StepStop`s/rejection entries. How about naming it `TransactionRejections`, `RejectionsBuilder` or simply `Rejections` (it's under the `transaction` package)? Then the call site could invoke e.g. `rejectionsBuilder.create(...)` or similar.

Reply:
Done, changed to `Rejections`.